includes/REST_API/Products_Controller.php

```diff
@@ -155,7 +155,14 @@ public function get_items( $request ) {
 		$query             = $this->shopping_vendors->get_vendor_class( $shopping_provider );
 
 		if ( ! $query ) {
-			return new WP_Error( 'unable_to_find_class', __( 'Unable to find class', 'web-stories' ), [ 'status' => 400 ] );
+			$shopping_provider = \strval( $this->settings->get_setting( Settings::SETTING_NAME_SHOPPING_PROVIDER ) );
+			if ( 'none' === $shopping_provider ) {
+				return new WP_Error( 'rest_shopping_provider', __( 'Please setup shopping provider', 'web-stories' ), [ 'status' => 400 ] );
```

Comment:
This check is a bit superfluous because someone could have filtered `web_stories_shopping_vendors` to have a custom solution.

The `$this->shopping_vendors->get_vendor_class()` call below already handles the `none` case.

Reply:
cc @spacedmonkey since you suggested that.

Reply:
If you filter this, why would you remove the none option? I think the two errors are different. One is if shopping vendor is not setup and one is for if not found. Better error messaging is better here.

Reply:
Ah, I forgot that the list of vendors is no longer hardcoded on the dashboard.

I suppose it should be fine then.